gcloud/resource_manager/iterator.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2014 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Iterators for paging through API responses.
+
+These iterators simplify the process of paging through API responses
+where the response is a list of results with a ``nextPageToken``.
+
+To make an iterator work, just override the ``get_items_from_response``
+method so that given a response (containing a page of results) it parses
+those results into an iterable of the actual objects you want::
+
+  class MyIterator(Iterator):
+      def get_items_from_response(self, response):
+          items = response.get('items', [])
+          for item in items:
+              my_item = MyItemClass(other_arg=True)
+              my_item._set_properties(item)
+              yield my_item
+
+You then can use this to get **all** the results from a resource::
+
+    >>> iterator = MyIterator(...)
+    >>> list(iterator)  # Convert to a list (consumes all values).
+
+Or you can walk your way through items and call off the search early if
+you find what you're looking for (resulting in possibly fewer
+requests)::
+
+    >>> for item in MyIterator(...):
+    >>>     print item.name
+    >>>     if not item.is_valid:
+    >>>         break
+"""
+
+from gcloud.resource_manager.project import Project
+
+
+class Iterator(object):
+    """A generic class for iterating through Cloud list responses.
+
+    :type client: :class:`gcloud.resource_manager.client.Client`
+    :param client: The client to use to make requests.
+
+    :type path: string
+    :param path: The path to query for the list of items.
+    """
+
+    PAGE_TOKEN = 'pageToken'
+    RESERVED_PARAMS = frozenset([PAGE_TOKEN])
+
+    def __init__(self, client, path, extra_params=None):
```

Comment:
This is the only part that differs from the storage base class. Since we don't intend it to be public facing, I suggest we bind a `connection` to it until the `client` pattern is implemented in storage.